python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -149,9 +181,17 @@ def evaluate(
 
         batch_count = 0
         df_grouped_stats = pd.DataFrame()
+        # defining variables to compute running mean and variance for t-test computations
+        agg_count, agg_mean, agg_M2 = 0, 0, 0
         for predictions_dict in test_dataset.map(_predict_fn).take(-1):
             predictions_df = pd.DataFrame(predictions_dict)
 
+            # Accumulating statistics for t-test calculation using 1/rank
+            clicked_records = predictions_df[predictions_df[self.feature_config.get_label("node_name")] == 1.0]
+            diff = (1/clicked_records[RankingConstants.NEW_RANK] - 1/clicked_records[
+                self.feature_config.get_rank("node_name")]).to_list()
+            agg_count, agg_mean, agg_M2 = compute_stats_from_stream(diff, agg_count, agg_mean, agg_M2)
```

Comment:
@mohazahran You should probably put this within an if block as well -> `if ttest_pvalue_threshold > 0:`